portfolio/src/main/java/com/google/sps/servlets/CommentService.java

```diff
@@ -0,0 +1,161 @@
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.time.ZoneId;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+
+/** 
+ * This class contains the fields and methods
+ * required to use the Comment feature.
+ */
+ public class CommentService {
+
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+   
+  public List<Comment> getComments(PreparedQuery results, int limit) {
+    Iterable<Entity> newComments = results.asIterable(FetchOptions.Builder.withLimit(limit));    
+    
+    /** /data page is updated to contain JSON file of all images up to limit. */
+    List<Comment> comments = results.asList(FetchOptions.Builder.withLimit(limit))
+        .stream()
+        .map(entity -> {
+            
+            String postedTime = (String) entity.getProperty("postedTime");
+            String title = (String) entity.getProperty("title");
+            String text = (String) entity.getProperty("text");
+            String imageUrl = (String) entity.getProperty("imageUrl");
+
+            return new Comment(postedTime, title, text, imageUrl);
+        })
+        .collect(Collectors.toList());
+    return comments;
+
+  }
+  
+  /** Adds the comment to the datastore. */
+  public void saveComment(HttpServletRequest request) {
+    ZonedDateTime dateTime =  ZonedDateTime.now().withZoneSameInstant(ZoneId.of("America/New_York"));
+
+    DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM, FormatStyle.MEDIUM);
+    String postedTime = dateTime.format(formatter) + " EST";
+
+    Entity commentEntity = new Entity("Comment");
+
+    commentEntity.setProperty("postedTime", postedTime);
+    commentEntity.setProperty("title", request.getParameter("title"));
+    commentEntity.setProperty("text", request.getParameter("text"));  
+    commentEntity.setProperty("imageUrl", getUploadedFileUrl(request, "image"));    
+    
+
+    datastore.put(commentEntity);
+  }
+
+ /**
+  * Removes Comments and Blob Information from datastore.
+  */
```

Comment:
Could we make this an inline javadoc? 

`/** Removes comments and blob information from datastore. */`

Reply:
Done.

Reply:
Done.